### DISCUSSION TITLE
the include shortcut doesn't render the md files well

### DISCUSSION THREAD

**Discussion by abdelkareemkobo** [quarto-dev/quarto-cli#6280] [Q&A]
### Description

![image](https://github.com/quarto-dev/quarto-cli/assets/72500236/7434e08a-9759-4dbe-b51d-28b5df07d333)
![image](https://github.com/quarto-dev/quarto-cli/assets/72500236/7bdf0e97-0024-4522-8e9c-85920f44061a)
The text copied from markdown files into index.qmd is not arranged well! how to solve this problem

**Comment by cscheid** [comment]
From our guide on bug reports:

The most helpful thing you can do to help us is to provide a minimal, self-contained, and reproducible example.

- minimal: This will often mean turning your large website project into a project with a single small document, and a single large .qmd file into a small (ideally, about 10-20 total lines of code) example. By doing this, you might also be able to learn more specifically what the problem is.
- self-contained: The more software dependencies we need to understand and install, the harder it is to track the bug down. As you reduce the code, remove as many dependencies as possible.
- reproducible: If we cannot run your example, we cannot track the bug down. Please make sure the file you submitted is enough to trigger the bug on its own.

**Reply by abdelkareemkobo** [reply]
sorry for misuse.
my problem is that using markdown files with include shortcut within qmd files.
the render post in my blog doesn't follow the markdown rules.
the whole content is made as one block the heading is not applied. like in the pictures one with from obsidian and one from the website.
this is not bug.
it's just something i don't know how to fix and there is no code to provide.
i searched quarto documents and found this in the a section with the name markdown output.
so i am asking what is best way to solve because i didn't understand the documents and seeking for you help.

**Reply by mcanouil** [reply]
You ran code to get the screenshot. We need a simpler version of that code to see what you saw on the screenshot.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by abdelkareemkobo** [reply]
````qmd 

---
title: "معاناة مع وسائل التواصل"
date: "2023-07-18"
categories: [life,personal]
format: html
draft: false
execute: 
  enabled: false
  code-fold: true
---
![](social_media_struggles.png)

trying to find if the problem of the writing is from obsidian or from quarto notebooks 
مرحبا أيها العالم كيف حالك:
		- أتمني أن تكون بأفضل حال يا عزيزي !

{{< include social_struggles.md >}}
````

**Reply by abdelkareemkobo** [reply]
This is the content of the social_struggles.md 
![image](https://github.com/quarto-dev/quarto-cli/assets/72500236/4511833c-042a-4fe9-b16b-51df8669010f)
This is the content of the post on the blog! how to fix the output and make it like the normal markdown format 
![image](https://github.com/quarto-dev/quarto-cli/assets/72500236/556d9b16-1248-4f26-8552-f3d7c2a221aa)

**Reply by cscheid** [reply]
Again, that's not the content of the file. That's a screenshot.

At this point, I will ask you to put all of your content in a git repository that we can access, and we won't be able to help you otherwise.

**Reply by mcanouil** [reply]
@abdelkareemkobo See the perfectly working example below (at least using the latest dev version):

<table>
<tr><th>Quarto document</th><th>Markdown document</th><th>HTML</th></tr>
<tr>
<td>

````qmd
---
title: "Quarto Playground"
engine: knitr
---

## Header 1

Some text.

{{< include markdown.md >}}
````

</td>
<td>

````qmd
### Header 2

Some other text with __markdown__ syntax.
````

</td>
<td>
<img width="579" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/ffbe36dc-9535-474b-9817-9e44bea18b5c">
</td>
</tr>
</table>

**Reply by abdelkareemkobo** [reply]
> Again, that's not the content of the file. That's a screenshot.
> 
> At this point, I will ask you to put all of your content in a git repository that we can access, and we won't be able to help you otherwise.

i send you a screenshot and the above one is the code

**Reply by mcanouil** [reply]
It's not complete!
Look at https://github.com/quarto-dev/quarto-cli/discussions/6280#discussioncomment-6500751
This is complete and actually working.

Also your partial code includes an image that we don't have and probably don't need.

**Reply by abdelkareemkobo** [reply]
You are right, there are no problems with Quarto and Markdown; it handles them fine. It was an oversight on my part, and I am really sorry for this. Here is the code to my repo. It works; there are two posts: one with English and code. It works as I want. The other one, called "social_struggles.md," and its index are not working as I need. The spaces and strips aren't applied correctly.

https://github.com/abdelkareemkobo/Abdelkareem-NNs

**Reply by cscheid** [reply]
I'm fairly sure your .md file has some problems. I'm not entirely positive because I'm not sure if my text editor is doing the right thing with the arabic text, but it looks like you have an extra space in the beginning of the first paragraph of the `# Intro` section. You can check that Pandoc itself isn't building your list correctly (I think that's what you mean by "spaces and strips", but I'm not sure).

```
$ cat social_struggles.md | quarto pandoc -t html
cat social_struggles.md | quarto pandoc -t html
<p>#blogging</p>
<h1 id="intro">Intro</h1>
<p>كان لي الكثير من المحاولات في الإختفاء التام من شبكات التواصل
الإجتماعي أولها - حذف حساب الفيس بوك لمده عام ونصف في فترة الثانوية
العامة بعد أن كان لدي حساب في عام 2011 عندما كان لدي حوالي 10 سنوات :) -
مرت الأيام واعدت إنشاء حساب في بداية الجامعة فقمت بحذفه بعد حوالي 6 شهور
عندما أحستت أني قد عجزت عن التأقلم مع هذا العالم الغريب والمريب الممتلئ
بالصور والسخرية وخصوصا الميمز. أن تصبح كل حياتك شخص يشارك صور مضحكه !
أحيانا يعجبني بعض منها ولكن اكثرها تافه أو منحط أو يحتاج نخبة من العلماء
لفك شفرة الضحك في تلك الصور الغريبه. ولست بقادر علي مشاركة صوري الخاصه
في كل مناسبة وإذا فعلت لا أستطيع الرد علي من يقول لي : أخويا! والي مني !
السندال، عم الناس, أبو الكواكب 💔 - الأغرب أن الأفكار علي المستوي اليومي
أصبحت تتشكل بناء علي نشاطي علي تلك الشبكات فتتبادر لي تلك المقولة
الرائعة التي ستجلب لي بعض الايكات والتعليقات وانني إذا نشرت كذا سيرد علي
فلان بكذا ..الخ أن أترك عقلي لهذا التيار المتخبط من المشاعر والأفكار
فمثل هذا الخبال لا أتحمل بَتاتًا. - والكثير الكثير من المعارك الصغيره
المتكرره الغير مفيده علي الأطلاق في الكرة والسياسية والهراء ..كان لي
ثلاث أسباب أتمسك بها حتي أظل في هذا المكان : - البقاء وسط العائلة
والأقارب ولكن هل هذه هي العلاقة التي أريد أن أتكون وسط عائلتي! - أحب
تشجيع المدرسين لي فلقد نالت علاقتي بيهم قدر كبير من مشاعري، فلهم كل
التقدير والإحترام وكلمة منهم ولو بسيطة عندي كوسام - سبب شخصي بعد الكثير
من التفكير وجودي الإفتراضي ربما يأخرني عن الواقعية التي أريدها في
حياتي</p>
<ul>
<li>حذفت المتصفح وتطبيق اليوتيوب وأصبحت أتابع ما أريده بإستخدام RSS</li>
<li>توتير أتابع عليه بعض الأشخاص ولكن لا أنشر عليه شئ تقريبا</li>
<li>العقيم linkedin أنتظر اللحظة التي أحصل علي وظيفة بدوام كامل حتي
أتخلص من هذا المكان المريب المصطنع.</li>
<li>أما بالنسبة للواتساب فلا أحب رؤية حالات الأخرين في الحقيقة الأمر يقف
بنسبة كبيره علي المجتمع المحيط ولكني لست راضي عنه فلهذا لدي هذا الأحتياط
الذي قد يبدو مبالغا للبعض. لا أريد رؤية ما يفكر به الناس, أما شخص يبكي
ويتألم لمجرد التألم وهذه “تلقح علي هذه” ليس لدي عقل لكل هذا.</li>
<li>تيليجرام مع إيقاف الإشعارت ووجوده علي الكمبيوتر وحذفه من الموبايل
فهو التطبيق الامثل عندي للتواصل مع الأشخاص خارج نطاق الدراسة أو
العمل</li>
<li>الأحب الي قلبي هو discord مناسب لكل شئ بدون عيوب في نظري التنظيم
والتركيز علي أعلي مستوي مقارنة بباقي التطبيقات</li>
</ul>
```

It's possible that pandoc itself is getting confused with RTL text, but I'm not sure. I do think the issue is that whatever tool you're using to generate this markdown isn't doing what you expect. If pandoc generates unexpected HTML from your markdown, then there's not much we'll be able to do.

**Comment by abdelkareemkobo** [comment]
you are right and this what i mean with spaces.
i generated this markdown with obsidian but with a plugin called auto rtl.
which i think is the cause of these errors.
i will write without using this plugin
